.github/actions/install-wdk/action.yml

```diff
@@ -130,16 +130,70 @@ runs:
           Write-Output "WindowsSdkBinPath=$packages\Microsoft.Windows.SDK.CPP.$version\c\bin" >> $env:GITHUB_ENV
         } else {
 
-          Write-Host "Using Winget to install WDK version $inputVersion..."
-          $wdk = "Microsoft.WindowsWDK.$inputVersion"
-          if ((Get-WinGetPackage -Id $wdk -Source winget -MatchOption Equals).Id -eq '$wdk') {
-            Write-Host "$wdk is already installed. Attempting to update..."
-            Update-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
-          } else {
-            Write-Host "Installing $wdk..."
-            Install-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
+        Write-Host "Using Winget to install WDK version $inputVersion..."
+        
+        # Import the WinGet PowerShell module
+        Write-Host "Importing WinGet PowerShell module..."
+        Import-Module Microsoft.WinGet.Client -Force
+        
+        # Detect and uninstall any existing Windows SDK packages to avoid conflicts
+        Write-Host "Detecting and removing existing Windows SDK packages..."
+        try {
+          $existingSDKs = Get-WinGetPackage | Where-Object { $_.Name -like "*Windows Software Development Kit*" -or $_.Id -like "Microsoft.WindowsSDK*" }
+        } catch {
+          Write-Host "No existing SDK packages found or error querying: ${_}"
+          $existingSDKs = @()
+        }
+        foreach ($sdk in $existingSDKs) {
+          Write-Host "Uninstalling existing SDK: $($sdk.Id) version $($sdk.InstalledVersion)"
+          try {
+            Uninstall-WinGetPackage $sdk.Id -Mode Silent -Force
+            Write-Host "Successfully uninstalled $($sdk.Id)"
+          } catch {
+            Write-Warning "Failed to uninstall $($sdk.Id): ${_}"
+          }
+        }
+        
+        # Detect and uninstall any existing WDK packages to avoid header conflicts
+        Write-Host "Detecting and removing existing WDK packages..."
+        try {
+          $existingWDKs = Get-WinGetPackage | Where-Object { $_.Name -like "*Windows Driver Kit*" -or $_.Id -like "Microsoft.WindowsWDK*" }
+        } catch {
+          Write-Host "No existing WDK packages found or error querying: ${_}"
+          $existingWDKs = @()
+        }
+        foreach ($wdk in $existingWDKs) {
+          Write-Host "Uninstalling existing WDK: $($wdk.Id) version $($wdk.InstalledVersion)"
+          try {
+            Uninstall-WinGetPackage $wdk.Id -Mode Silent -Force
+            Write-Host "Successfully uninstalled $($wdk.Id)"
+          } catch {
+            Write-Warning "Failed to uninstall $($wdk.Id): ${_}"
           }
+        }
+        
+        # Install the matching Windows SDK first
+        $sdkPackageId = "Microsoft.WindowsSDK.$sdkVersion"
+        Write-Host "Installing matching Windows SDK: $sdkPackageId..."
+        try {
+          Install-WinGetPackage -Id $sdkPackageId -Source winget -MatchOption Equals -Mode Silent -Force
```

Comment:
pass `-override '/q /features +' explicitly only for sdkVersion==22621 because of bug.